misk/src/main/kotlin/misk/web/jetty/JettyService.kt

```diff
@@ -96,16 +109,34 @@ class JettyService @Inject internal constructor(
       val httpsConfig = HttpConfiguration(httpConfig)
       httpsConfig.addCustomizer(SecureRequestCustomizer())
 
-      val alpn = ALPNServerConnectionFactory("h2", "http/1.1")
+      val ssl = SslConnectionFactory(sslContextFactory, "alpn")
+      httpsConnectionFactories += ssl
+
+      val alpnProtocols = if (webConfig.http2) listOf("h2", "http/1.1") else listOf("http/1.1")
+      val alpn = ALPNServerConnectionFactory(*alpnProtocols.toTypedArray())
       alpn.defaultProtocol = "http/1.1"
-      val ssl = SslConnectionFactory(sslContextFactory, alpn.protocol)
-      val http2 = HTTP2ServerConnectionFactory(httpsConfig)
-      if (webConfig.jetty_max_concurrent_streams != null) {
-        http2.maxConcurrentStreams = webConfig.jetty_max_concurrent_streams
+      httpsConnectionFactories += alpn
+
+      if (webConfig.http2) {
+        val http2 = HTTP2ServerConnectionFactory(httpsConfig)
+        if (webConfig.jetty_max_concurrent_streams != null) {
+          http2.maxConcurrentStreams = webConfig.jetty_max_concurrent_streams
+        }
+        httpsConnectionFactories += http2
       }
+
       val http1 = HttpConnectionFactory(httpsConfig)
-      val httpsConnector = ServerConnector(server, null, null, null,
-          webConfig.acceptors ?: -1, webConfig.selectors ?: -1, ssl, alpn, http2, http1)
+      httpsConnectionFactories += http1
+
+      val httpsConnector = ServerConnector(
```

Comment:
This `startUp` method is long. I would move this line to a `createHttpsConnector` method (same for `createHttpConnector`)

Reply:
Lemme consider for follow-up. I’m not bothered by long plumbing methods.